### ISSUE TITLE
Incorrect handling of number input results in overflow and underflow in Create Auction dialog

### ISSUE THREAD

**Issue by tannyl** [anegostudios/VintageStory-Issues#4617]
### Game Version

1.20.0-rc.1

### Platform

Linux

### Modded

Vanilla

### SP/MP

Singleplayer

### Description

In the **Create Auction** dialog we have an input field, where you can type in the price in rusty gears you wish to sell an item for.

This field does not have max and min limits defined for its input so you can type any arbitrary number and for that matter letters. You can even enter negative numbers which does not make any sense from a business standpoint. :smile:

This can results in things like overflow and underflow situations when using the input fields value these places:
- https://github.com/anegostudios/vssurvivalmod/blob/f905b60d6967cf9758689df723886230205b7260/Systems/Auction/GuiDialogCreateAuction.cs#L102
This section updates the **Trader cut on sale** value. The number is retrieved via
`int cost = Composers["tradercreateauction"].GetNumberInput("price").GetText().ToInt(1)`
If the value in the input field is lager then `+2,147,483,647 (Int32 max)` the trader cut value ends up as `0`.
If the value in the input field is smaller then `-2,147,483,648 (Int32 min)` the trader cut value ends up as `0`.
If the input value is incorrect (not a number) the trader cut value ends up as `0`.
I guess this happens due to `ToInt(1)` using `int.TryParse` internally which fails and `cost` ends up as `0`.

- https://github.com/anegostudios/vssurvivalmod/blob/f905b60d6967cf9758689df723886230205b7260/Systems/Auction/GuiDialogCreateAuction.cs#L124
This section tries to create the auction. The price number is retrieved via
`int price = (int)cmp.GetNumberInput("price").GetValue();` 
The price is returned as a float and then converted to an int, which is not how it was handled in the previous section. This results in some more strange behavior.
If the value in the input field is larger then `+2,147,483,583` the price ends up as `0` and creating the auction fails with an error.
If the value in the input field is `<= 0` the price is `0` or less and the auction ends up failing.
If the input value is incorrect (not a number) the price most likely ends up as `0` as well and creating the auction fails.

Also the price shown in the lists of auctions after you managed to create one, might end up incorrect (either just visually or actually stored incorrectly behind the scenes) if you use really large numbers. I could not find a reason for this yet, but I suspect it might be due to using floats somewhere in the code. As an example if I sell something for `50,000,473` the price shown in the list is actually `50,000,472`. If I use `50,000,395` the price ends up at `50,000,396`. At even higher prices it gets even more wonky.

So to summarize:
I think the price input field in the Create Action dialog needs to have an lower and upper limit set, as well as limit it to numbers only. And maybe the code calculating the price for the auction entries needs a look through.

### Logs

```Log```
